rotkehlchen/api/rest.py

```diff
@@ -4413,3 +4417,63 @@ def reprocess_skipped_external_events(self) -> Response:
             result=_wrap_in_ok_result({'total': total, 'successfull': successfull}),
             status_code=HTTPStatus.OK,
         )
+
+    def export_history_events(
+            self,
+            filter_query: HistoryBaseEntryFilterQuery,
+            directory_path: Optional[Path],
+    ) -> Response:
+        dbevents = DBHistoryEvents(self.rotkehlchen.data.db)
+        with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
+            history_events, _, _ = dbevents.get_history_events_and_limit_info(
+                cursor=cursor,
+                filter_query=filter_query,
+                has_premium=True,
+                entries_limit=None,
+            )
+
+        if len(history_events) == 0:
+            return api_response(
+                wrap_in_fail_result('No history processed in order to perform an export'),
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+        serialized_history_events = []
+        headers: dict[str, None] = {}
+        for event in history_events:
+            serialized_event = event.serialize_for_csv()
+            serialized_history_events.append(serialized_event)
+            # maintain insertion order without storing extra info
+            headers.update({key: None for key in serialized_event})
+
+        try:
+            with tempfile.TemporaryDirectory() as temp_dir:
+                # on download
+                if directory_path is None:
+                    file_path = Path(temp_dir) / FILENAME_HISTORY_EVENTS_CSV
```

Comment:
mm this works I believe because Path calls str on the object or something like this but can you use

```suggestion
                    file_path = Path(temp_dir.name) / FILENAME_HISTORY_EVENTS_CSV
```

for me at least is clearer